src/main/java/edu/cornell/kfs/module/ar/service/impl/CuCustomerAddressHelperServiceImpl.java

```diff
@@ -0,0 +1,109 @@
+    
+    /*
+     * CU Customization: KFSPTS-26393
+     * When creationProcessType is BATCH, validate that the customer number on the award as part if the customer
+     * address composite key matches the customer number associated to the agency associated to the award.
+     */
+    public boolean agencyCustomerMatchesAwardCustomer(ContractsAndGrantsBillingAward award,
+            ContractsAndGrantsInvoiceDocumentCreationProcessType creationProcessType) {
+        
+        if (ContractsAndGrantsInvoiceDocumentCreationProcessType.BATCH == creationProcessType) {
+            if (StringUtils.isNotBlank(obtainCustomerNumberFromAwardAgencyCustomer(award)) &&
+                    StringUtils.isNotBlank(award.getCustomerNumber()) &&
+                    obtainCustomerNumberFromAwardAgencyCustomer(award).equalsIgnoreCase(award.getCustomerNumber())) {
+                return true;
+            } else {
+                LOG.info("agencyCustomerMatchesAwardCustomer: Mismatched Customer records detected for " +
+                         "Award.Agency and Award.CustomerAddress : Award.Agency.CustomerNumber = " +
+                         (StringUtils.isBlank(obtainCustomerNumberFromAwardAgencyCustomer(award)) ? null : obtainCustomerNumberFromAwardAgencyCustomer(award)) +
+                         " Award.CustomerNumber = " + award.getCustomerNumber() +
+                         " Award.CustomerAddressIdentifier = " +
+                         (ObjectUtils.isNull(award.getCustomerAddressIdentifier()) ? "null" : award.getCustomerAddressIdentifier().intValue()));
+                return false;
+            } 
+        } else {
+            LOG.info("agencyCustomerMatchesAwardCustomer: Validation bypassed due to creationProcessType being " + creationProcessType.getName());
+            return true;
+        }
+    }
+    
+    /*
+     * CU Customization: KFSPTS-26393
+     * When creationProcessType is BATCH, validate that a customer address exists in the same manner that it is
+     * obtained by downstream processing in ContractsGrantsInvoiceCreateDocumentServiceImpl.buildInvoiceAddressDetails.
+     */
+    public boolean invoicingCustomerAddressExists(ContractsAndGrantsBillingAward award,
+            ContractsAndGrantsInvoiceDocumentCreationProcessType creationProcessType) {
+        
+        CustomerAddress customerAddress = null;
+        
+        if (ContractsAndGrantsInvoiceDocumentCreationProcessType.BATCH == creationProcessType) {
+            if (ObjectUtils.isNotNull(award.getCustomerAddressIdentifier()) &&
+                    ObjectUtils.isNotNull(award.getCustomerNumber())) {
+                customerAddress = customerAddressService.getByPrimaryKey(award.getCustomerNumber(),
+                        award.getCustomerAddressIdentifier());
+            }
+            if (ObjectUtils.isNull(customerAddress) && 
+                    ObjectUtils.isNotNull(award.getCustomerNumber())) {
```

Comment:
Same here, should this check be updated to use StringUtils.isNotBlank to ensure it checks on whitespaces too? Sorry for not asking this earlier.

Reply:
Good catch, updating.